### PR TITLE
Particle filter example bug fix and refactoring

### PR DESCRIPTION
**Pull request by THargreaves**

### CODE REVIEW

File: examples/smc.jl [hunk 1]
```diff
@@ -106,14 +105,14 @@ isdone(t, ::Nothing) = false
...
+        x[t + 1] = rand(rng, Normal(x[t], params.u))
     end
 end
 
 samples = sweep!(rng, fill(Particle(x[1]), N), systematic_resampling)
 traces = reverse(hcat(map(linearize, samples)...))
 
-scatter(traces; color=:black, opacity=0.3)
-plot!(x)
+scatter(traces; color=:black, opacity=0.3, label="")
```
Comment (FredericWantiez) [inline examples/smc.jl:117]: `scatter(traces; color=:black, opacity=0.3, label=False)`

File: examples/smc.jl [hunk 2]
```diff
@@ -106,14 +105,14 @@ isdone(t, ::Nothing) = false
 x, observations = zeros(T), zeros(T)
 x[1] = rand(rng, Normal(0, 1))
 for t in 1:T
-    observations[t] = rand(rng, Normal(x[1], params.v^2))
+    observations[t] = rand(rng, Normal(x[t], params.v))
     if t < T
-        x[t + 1] = rand(rng, Normal(x[t], params.u^2))
+        x[t + 1] = rand(rng, Normal(x[t], params.u))
```
Comment (FredericWantiez) [inline examples/smc.jl:110]: They should be flipped ? 
```julia
for t in 1:T
    observations[t] = rand(rng, Normal(x[t], params.u))
    if t < T
        x[t + 1] = rand(rng, Normal(x[t], params.v))
    end
end
```

**Comment by FredericWantiez** [comment]
But thanks so much ! It was on my todo list to fix it since I wrote it a bit too fast

File: examples/smc.jl [hunk 3]
```diff
@@ -106,14 +105,14 @@ isdone(t, ::Nothing) = false
 x, observations = zeros(T), zeros(T)
 x[1] = rand(rng, Normal(0, 1))
 for t in 1:T
-    observations[t] = rand(rng, Normal(x[1], params.v^2))
+    observations[t] = rand(rng, Normal(x[t], params.v))
```
Comment (FredericWantiez) [inline examples/smc.jl]:
```suggestion
    observations[t] = rand(rng, Normal(x[t], params.u))
```

File: examples/smc.jl (hunk 2 above)
Comment (FredericWantiez) [inline examples/smc.jl]:
```suggestion
        x[t + 1] = rand(rng, Normal(x[t], params.v))
```

**Comment by THargreaves** [comment]
Thanks for the suggestions. Both made and committed.

**Review by FredericWantiez — APPROVED** [review]
Thanks !

**Pull request merged**